libs/SmartStore/SmartStore/Classes/SFSmartStoreDatabaseManager.m

```diff
@@ -199,12 +199,12 @@ + (FMDatabase*) setKeyForDb:(FMDatabase*) db key:(NSString *)key salt:(NSString
 
 + (FMDatabase*) unlockDatabase:(FMDatabase*)db key:(NSString*)key salt:(NSString *)salt {
     if ([db open]) {
-        // Using sqlcipher 2.x kdf iter because 3.x default (64000) and 4.x default (256000) are too slow
-        [[db executeQuery:@"PRAGMA cipher_default_kdf_iter = 4000"] close];
-       
         if (key)
            [db setKey:key];
         
+        // Using sqlcipher 2.x kdf iter because 3.x default (64000) and 4.x default (256000) are too slow
```

Comment:
Should it happen before setKey?

Reply:
@wmathurin The [docs explicitly say](https://www.zetetic.net/sqlcipher/sqlcipher-api/#kdf_iter) it should happen after setting the key: 

> PRAGMA kdf_iter must be called after PRAGMA key and before the first actual database operation or it will have no effect.

Reply:
👍 
That makes sense. cipher_default_kdf_iter being "global" could happen earlier.